### DISCUSSION TITLE
Set up a startup config

### DISCUSSION THREAD

**Discussion by dmartinezgarciaCPG** [glzr-io/glazewm#244] [Q&A]
Hey! I've installed GlazeWM and it seems pretty amazing and powerfull.

I was wondering if I can, in some way, create some rules when starting up the computer.
My goal is to turn on the computer and launch a bunch of programs and move those programs to a different workspaces, resize them... at the end of the day just automatize the setup that I do manually every time I use the computer.

Thanks a lot and have a nice day!

**Comment by Okabe9** [comment]
I'm also interested in this. I used Bspwm and I loved the option to bind process to workspace so everytime i opened Chrome i knew it was inside Workspace 2 and VS Code in 1 for example. Don't know if there's anyway to do that with GlazeWM.

**Reply by Okabe9** [reply]
Okay, I've made some more investigation and hope this could help. 

To open the programs on startup i didn't found anything inside GlazeVM that could help. But that's pretty easy using windows Task Scheduler: 
Here you have an example: https://www.wintips.org/how-to-start-a-program-at-startup-with-task-scheduler/

After doing this with each program you want to run on startup you would've to add the commands to each matching process like this: 

```
- command: "move to workspace 1"
    match_process_name: "Code"
- command: "move to workspace 2"
    match_process_name: "chrome"
- command: "move to workspace 3"
    match_process_name: "Obsidian"
- command: "move to workspace 4"
    match_process_name: "Everything"
- command: "move to workspace 5"
    match_process_name: "powershell"
```
    
 You can setup multiple commands using instead: 
 
` ["move to workspace 2", "focus workspace 2"]`

**Comment by MyGitHubBlueberry** [comment]
For programs to work on startup either add reference to them in your `C:\Users\username\AppData\Roaming\Microsoft\Windows\Start Menu\Programs` folder or search for 'startup' in windows settings (`Windows+I` to open),  and enable required programs to launch on start.
